### ISSUE TITLE
login cracker segfault

### ISSUE THREAD

**Issue by gainan** [cSploit/android#236]
When launching the login cracker module against a target, it crashes (android 4.4.4 and 5.x).

Executing hydra from the shell segfaults:
root@lg:/data/data/org.csploit.android/files/tools/hydra # ./hydra
./hydra
[1] + Stopped (signal)     ./hydra
root@lg:/data/data/org.csploit.android/files/tools/hydra #

I´ve compiled the latest version of thc-hydra https://github.com/vanhauser-thc/thc-hydra/, and works ok.

**Comment by tux-mind** [comment]
Hi @gainan thanks for finding another bug :blush: 

how did you compiled hydra ? using `cSploit/jni/build.sh` ?

thanks in advance :blush:

**Comment by gainan** [comment]
no, I just crosscompiling it using arm-linux-gnueabi*, but without libs! I compiled it to only ensure me that it was not another problem.

**Comment by tux-mind** [comment]
probably the problem is in using some of those libraries.
btw share a logcat taken after the segfault, it should contain a native stacktrace.

**Comment by gainan** [comment]
This bug has gone away with core tools 1.0.6! it doesn't crash anymore.

**Comment by tux-mind** [comment]
mmmmm... it's quite crazy, i changed only nmap stuff from `1.0.5` to `1.0.6`.
feel free to reopen it if if happens again :blush:

**Comment by gainan** [comment]
This problem is not solved for some(rubenoo for example). ACRAC caught this error:
>61	10-09 15:30:20.404 D/CSPLOIT[core.ChildManager.async](10208): { handler='hydra', cmd='-F -l admin -x \"1:1:a\" -s 23 -V -t 10 192.168.2.1 telnet' } => 11
62	10-09 15:30:20.416 D/CSPLOIT[core.ChildManager.onEvent](10208): received an event: StderrNewline: { line='CANNOT LINK EXECUTABLE: cannot locate symbol \"PKCS7_free\

**Comment by gainan** [comment]
Same problem reported by @uranium-waffles:
>10-14 19:17:16.062 D/CSPLOIT[core.ChildManager.async](27208): { handler='hydra', cmd='-F -l admin -x "1:1:a"  -s 80 -V -t 10 192.168.1.1 http-head /' } => 9
10-14 19:17:16.082 F/libc    (27393): Fatal signal 11 (SIGSEGV), code 1, fault addr 0x4 in tid 27393 (hydra)

**Comment by tux-mind** [comment]
```
/media/data/documents/programs/csploit/android/cSploit/jni/firebird/src/common/classes/alloc.cpp:520
/media/data/documents/programs/csploit/android/cSploit/jni/firebird/src/include/../common/classes/alloc.h:334
```
